### ISSUE TITLE
Phase 6: ML Training Integration + AI Self-Play Loop

### ISSUE THREAD

**Issue by KoalaTrapLord** [KoalaTrapLord/commander-ai-lab#173]
### Overview
Close the training loop. Use game logs produced by Phase 2 AI decisions as training data for the PPO reinforcement learning pipeline. Enable headless self-play: all 4 players are AI-driven, no Unity required, runs overnight for batch training data generation.

### Dependencies
- Phases 1–3 must be complete (Forge bridge + AI engine + WebSockets operational)

#### Decision Logging (add to Phase 2 as needed)
- [ ] `commander_ai_lab/ai/decision_logger.py`
  - Log per-decision: `game_id`, `turn`, `phase`, `player_id`, `personality`, `brain_used`, `prompt`, `raw_response`, `parsed_action`, `action_was_valid`, `fallback_used`
  - Store to `data/decision_logs/{game_id}.jsonl`
  - Configurable via `LOG_AI_DECISIONS=true` env var

#### Game Result Logging
- [ ] `commander_ai_lab/forge/game_result_logger.py`
  - On Forge game end: capture winner, turn count, life totals, eliminations, commander damage dealt
  - Log to `data/game_results.jsonl`
  - Trigger `TrainingDataBuilder` after each game

#### Training Data Builder
- [ ] `commander_ai_lab/training/training_data_builder.py`
  - Convert `decision_log + game_result` pairs → `(state, action, reward)` tuples
  - Reward signal: +1.0 winner, +0.0 elimination order 2, -0.25 elimination order 3, -0.5 first eliminated
  - Export to `data/training_sets/{timestamp}_training.pkl`

#### Headless Self-Play Runner
- [ ] `scripts/run_self_play.py`
  - Accepts: `--num-games`, `--deck-pool-path`, `--output-dir`, `--seed`
  - Randomly samples 4 decks per game from pool
  - Launches `ForgeGameSession`, runs all 4 AI players to completion
  - No Unity needed — FastAPI WebSocket not used in headless mode
  - Progress bar via `tqdm`, estimated ETA per batch
  - Example: `python scripts/run_self_play.py --num-games 500 --deck-pool-path data/decks/`

#### PPO Integration
- [ ] Hook `training_data_builder` output into existing `ml_pipeline/ppo_trainer.py`
  - Confirm observation space includes: hand cards (encoded), battlefield state, life totals, mana pool, phase
  - Confirm action space encodes: cast spell (card slot), play land, attack target, pass priority
  - Add `--from-self-play` flag to trainer to load from `data/training_sets/`

### Verification
```bash
# Run 10 headless self-play games
python scripts/run_self_play.py --num-games 10 --deck-pool-path data/decks/ --seed 42

# Confirm logs generated
ls data/decision_logs/
ls data/game_results.jsonl
ls data/training_sets/

# Run PPO training on self-play data
python ml_pipeline/ppo_trainer.py --from-self-play --epochs 10
```

### Acceptance Criteria
- 100 headless self-play games run without human intervention or Unity
- `decision_logs/` contains per-turn prompt/response for every AI decision
- `game_results.jsonl` captures winner, elimination order, turn count for every game
- Training data exported as `(state, action, reward)` tuples consumable by PPO pipeline
- PPO trainer accepts self-play data and runs without errors
